### ISSUE TITLE
Issue with Integrating a New Loss Function into Knowledge Distillation Framework

### ISSUE THREAD

**Issue by PietroManganelliConforti** [DefangChen/SimKD#16]
Hi,

We're encountering difficulties incorporating a new loss function into the primary KD ones implemented within your repository. Despite adding it to the existing loss calculation within the file "/helper/loops.py", we've noticed no discernible change in the output.

We have the presumption that our approach lacks some essential elements necessary for proper implementation in pytorch or in the repository code. Any insights or suggestions on how to proceed would be very helpful.

> Criterion = Nn.MSEloss()
> 
> Loss_custom = criterion( torch.tensor(map1, requires_grad=True),
> torch.tensor(map2, requires_grad=True)).cuda()

and then,

> loss = opt.cls * loss_cls + opt.div * loss_div + opt.beta * loss_kd + opt.custom_weight * Loss_custom

Map1 and map2 are numpy.ndarray(), while custom_weight is a float.

Thanks in advance!

edit: I've found the error, the added function for the maps wasn't planned to be differentiable and a quick-fix created a theoretical error. Thanks anyways.

**Comment by ptrblck** [comment]
Cross-post from [here](https://discuss.pytorch.org/t/add-a-custom-loss-in-a-public-github-repository-distillation/198504/2?u=ptrblck).
